### PR TITLE
update developer guide for ks 3.0.0

### PR DESCRIPTION
**Pull request by yuswift**
Please help me review the developer guide for ks v3.0.0 
@zryfish 
@FeynmanZhou

### CODE REVIEW

File: developer-guide/development/development-workflow.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ Create a branch from master:
 git checkout -b myfeature
 ```
 
-Then edit code on the `myfeature` branch.
+Then edit code on the `myfeature` branch. You can refer [effective_go](https://golang.org/doc/effective_go.html) while coding.
```
Comment (FeynmanZhou) [inline developer-guide/development/development-workflow.md:55]: refer => refer to

**Comment by FeynmanZhou** [comment]
@zryfish

File: developer-guide/development/kubesphere-config-example.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  oauthOptions:
+    accessTokenMaxAge: 0
+ldap:
+  host: openldap.kubesphere-system.svc:389
+  managerDN: cn=admin,dc=kubesphere,dc=io
+  managerPassword: admin
+  userSearchBase: ou=Users,dc=kubesphere,dc=io
+  groupSearchBase: ou=Groups,dc=kubesphere,dc=io
+redis:
+  host: 10.233.34.251
```
Comment (zryfish) [inline developer-guide/development/kubesphere-config-example.yaml:23]: use service name instead

File: developer-guide/development/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+See [how to install kubesphere](https://kubesphere.io/docs/).
+
+## Hello KubeSphere
+```bash
+git clone https://github.com/kubesphere/kubesphere.git
+cd kubesphere && mkdir -p kapis/hellokubesphere/v1alpha1 && cd kapis/hellokubesphere/v1alpha1
+```
+Create a file named hander.go. In this tutorial, it only returned a http response with "hello-kubesphere", 
+you can also write your custom logic.  
+```bash
```
Comment (zryfish) [inline developer-guide/development/quickstart.md:46]: syntax highlight => go

File: developer-guide/development/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+Ks-apiserver is based on [go-restful](https://github.com/emicklei/go-restful). This document explains how to develop api with ks-apiserver from scratch.
+If you are interested in developing controller-manager, see [sample-controller](https://github.com/kubernetes/sample-controller)
+and [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder).
+
+## Prerequisites
+
+### Go
+
+KubeSphere development is based on [Kubernetes](https://github.com/kubernetes/kubernetes). Both of them are written in [Go](http://golang.org/). If you don't have a Go development environment, please [set it up](http://golang.org/doc/code.html) first.
```
Comment (zryfish) [inline developer-guide/development/quickstart.md:11]: KubeSphere is based on

File: developer-guide/development/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+Put 
+
+```bash
+urlruntime.Must(hellokubesphere.AddToContainer(s.container))
+```
+at the end of installKubeSphereAPIs function. <br>
+Kubesphere has strict access control, for convenience, you can modify the buildHandlerChain function in  pkg/apiserver/apiserver.go  and change excludedPaths variable like this
+
+```
+excludedPaths := []string{"/oauth/*", "/kapis/config.kubesphere.io/*", "/kapis/version", "/kapis/example.kubesphere.io/v1alpha1/hello-kubesphere"}
```
Comment (zryfish) [inline developer-guide/development/quickstart.md:116]: Don't add example to exclueded api pathes, it's only authentication and authorization apis

**Comment by FeynmanZhou** [comment]
@yuswift Please fix the conflicts.

**Comment by rayzhou2017** [comment]
miss sign off. Please resolve the conflicts and conversation if update already done according to the comments. @yuswift

**Pull request closed without merging**